routes/utils.js

```diff
@@ -30,35 +30,57 @@ router.get('/search', (req, res, next) => {
     // search for item name with given string
     if (results.length <= 0) {
       // get from database
+      let ids = conHelper.getObjectIdByItemName(req.query.itemName);
+      for(let i = 0; i < ids.length; i++){
+        results[i] = conHelper.getObjectById(ids[i]);
+      }
+      return res.status(200).json(results);
+
     } else {
       // filter by results
       results = results.filter(val => val.itemName.toLowerCase()
         .includes(req.query.itemName.toLowerCase()));
+      return res.status(200).json(results);
     }
   }
 
   if (req.query.zipcode) {
     if (results.length <= 0) {
       // get from database
+      let ids = conHelper.getObjectIdByZipcode(req.query.zipcode);
```

Comment:
So for all of these loops- can't we issue one statement to just get all the objects with that zipcode, rather than getting all the ids and then getting all the objects?  This seems like it'll be way slower since there's so many queries